### PR TITLE
Replace DefaultTextWatcher with extensions in core-ktx

### PR DESCRIPTION
**Pull request by Goooler**
https://github.com/androidx/androidx/blob/91003410072c031acf535628b04a456f17ff9c61/core/core-ktx/src/main/java/androidx/core/widget/TextView.kt#L23-L96

### CODE REVIEW

File: app/src/main/java/com/keylesspalace/tusky/ListsActivity.kt [hunk 1]
```diff
@@ -143,8 +143,8 @@ class ListsActivity : BaseActivity(), Injectable, HasAndroidInjector {
             .show()
 
         val positiveButton = dialog.getButton(Dialog.BUTTON_POSITIVE)
-        editText.onTextChanged { s, _, _, _ ->
-            positiveButton.isEnabled = s.isNotBlank()
+        editText.doOnTextChanged { s, _, _, _ ->
+            positiveButton.isEnabled = s.isNullOrBlank()
```
Comment (nikclayton) [inline app/src/main/java/com/keylesspalace/tusky/ListsActivity.kt:147]: Doesn't this invert the meaning, compared to the previous code?

Comment (Goooler) [reply app/src/main/java/com/keylesspalace/tusky/ListsActivity.kt:147]: Previous:
https://github.com/tuskyapp/Tusky/blob/e98925b42687fdf0a6eaf098231babdbcc802d48/app/src/main/java/com/keylesspalace/tusky/util/ViewExtensions.kt#L50-L58

Now:
https://github.com/androidx/androidx/blob/91003410072c031acf535628b04a456f17ff9c61/core/core-ktx/src/main/java/androidx/core/widget/TextView.kt#L42-L49

The `CharSequence` param is nullable now.

Comment (Lakoja) [reply app/src/main/java/com/keylesspalace/tusky/ListsActivity.kt:147]: I would also give it an additional !

With this the create list button is only enabled when the name is empty.

Comment (Goooler) [reply app/src/main/java/com/keylesspalace/tusky/ListsActivity.kt:147]: I got you mean, just a typo here.

Comment (Goooler) [reply app/src/main/java/com/keylesspalace/tusky/ListsActivity.kt]: a2f47cf7094dff65d6c692162b241f72609697e5

File: app/src/main/java/com/keylesspalace/tusky/components/compose/ComposeActivity.kt [hunk 2]
```diff
@@ -393,8 +394,8 @@ class ComposeActivity :
 
         val mentionColour = binding.composeEditField.linkTextColors.defaultColor
         highlightSpans(binding.composeEditField.text, mentionColour)
-        binding.composeEditField.afterTextChanged { editable ->
-            highlightSpans(editable, mentionColour)
+        binding.composeEditField.doAfterTextChanged { editable ->
+            highlightSpans(editable!!, mentionColour)
```
Comment (Goooler) [inline app/src/main/java/com/keylesspalace/tusky/components/compose/ComposeActivity.kt:398]: https://github.com/tuskyapp/Tusky/blob/cfea5700b0ba652de580b9d78e848fc280079c7b/app/src/main/java/com/keylesspalace/tusky/util/SpanUtils.kt#L138-L143

File: app/src/main/java/com/keylesspalace/tusky/components/account/AccountActivity.kt [hunk 3]
```diff
@@ -645,21 +644,14 @@ class AccountActivity : BottomSheetActivity(), ActionButtonActivity, HasAndroidI
                 subscribing = relation.subscribing
         }
 
-        // remove the listener so it doesn't fire on non-user changes
-        binding.accountNoteTextInputLayout.editText?.removeTextChangedListener(noteWatcher)
```
[the remark this answers is not in the document]
Comment (connyduck) [reply app/src/main/java/com/keylesspalace/tusky/components/account/AccountActivity.kt:649]: But now it is called on non-user changes.
Pull to refresh -> unnecessary api call to `/api/v1/accounts/:id/note` is sent

[the remark this answers is not in the document]
Comment (Goooler) [reply app/src/main/java/com/keylesspalace/tusky/components/account/AccountActivity.kt]: b60675cff9c253bce3ee1ba4481e260acc5fb490

**Review by connyduck — APPROVED** [review]

**Pull request merged**